modules/settings/settings.php

```diff
@@ -7,119 +7,79 @@
-		add_action( 'admin_init', array( $this, 'register_settings' ) );
+		add_action( 'admin_init', [ __CLASS__, 'helper_settings_validate_and_save' ], 100 );
+		add_action( 'admin_init', [ __CLASS__, 'register_settings' ] );
 
-		add_action( 'admin_print_styles', array( $this, 'action_admin_print_styles' ) );
-		add_action( 'admin_print_scripts', array( $this, 'action_admin_print_scripts' ) );
-		add_action( 'admin_enqueue_scripts', array( $this, 'action_admin_enqueue_scripts' ) );
+		add_action( 'admin_enqueue_scripts', [ __CLASS__, 'action_admin_enqueue_scripts' ] );
 	}
 
 	/**
 	 * Add the settings page to the admin menu
 	 */
-	public function add_admin_menu() {
+	public static function add_admin_menu(): void {
 		$menu_title = __( 'Settings', 'vip-workflow' );
 
-		add_submenu_page( Custom_Status::SETTINGS_SLUG, $menu_title, $menu_title, 'manage_options', self::SETTINGS_SLUG, [ $this, 'render_settings_view' ] );
+		add_submenu_page( CustomStatus::SETTINGS_SLUG, $menu_title, $menu_title, 'manage_options', self::SETTINGS_SLUG, [ __CLASS__, 'render_settings_view' ] );
 	}
 
 	/**
-	 * Add settings JS to the settings page
+	 * Enqueue resources that we need in the admin settings page
+	 *
+	 * @access private
 	 */
-	public function action_admin_enqueue_scripts() {
-		if ( VIP_Workflow::is_settings_view_loaded( self::SETTINGS_SLUG ) ) {
-			wp_enqueue_script( 'vip-workflow-settings-js', $this->module_url . 'lib/settings.js', array( 'jquery' ), VIP_WORKFLOW_VERSION, true );
+	public static function action_admin_enqueue_scripts(): void {
+		if ( HelperUtilities::is_settings_view_loaded( self::SETTINGS_SLUG ) ) {
+			$asset_file = include VIP_WORKFLOW_ROOT . '/dist/modules/settings/settings.asset.php';
+			$dependencies = [ ...$asset_file['dependencies'], 'jquery' ];
+			wp_enqueue_script( 'vip-workflow-settings-js', VIP_WORKFLOW_URL . 'dist/modules/settings/settings.js', $dependencies, $asset_file['version'], true );
 		}
 	}
 
-	/**
-	 * Add settings styles to the settings page
-	 */
-	public function action_admin_print_styles() {
-		wp_enqueue_style( 'vip_workflow-settings-css', $this->module_url . 'lib/settings.css', false, VIP_WORKFLOW_VERSION );
-	}
-
-	/**
-	 * Extra data we need on the page for transitions, etc.
-	 */
-	public function action_admin_print_scripts() {
-		?>
-		<script type="text/javascript">
-			var vw_admin_url = '<?php echo esc_url( get_admin_url() ); ?>';
-		</script>
-			<?php
-	}
-
 	/**
 	 * Register the settings for the module
 	 */
-	public function register_settings() {
-		add_settings_section( $this->module->options_group_name . '_general', false, '__return_false', $this->module->options_group_name );
+	public static function register_settings(): void {
+		add_settings_section( OptionsUtilities::get_module_options_general_key( self::SETTINGS_SLUG ), false, '__return_false', OptionsUtilities::get_module_options_key( self::SETTINGS_SLUG ) );
```

Comment:
Can we precompute these in variables and shorten these lines a bit?

```suggestion
		$settings_section_id = OptionsUtilities::get_module_options_general_key( self::SETTINGS_SLUG );
		$settings_section_page = OptionsUtilities::get_module_options_key( self::SETTINGS_SLUG );
		
		add_settings_section( $settings_section_id, false, '__return_false', $settings_section_page );
```

We may also just remove `self::SETTINGS_SLUG` entirely if it's always the only parameter [like this other comment](https://github.com/Automattic/vip-workflow-plugin/pull/59#discussion_r1792544850).

Reply:
Yip, done this too. I left the slug as a default in the parameter as the cleanup method that we use in the tests utilizes it.